packages/core/src/builder.class.ts

```diff
@@ -1104,9 +1104,15 @@ export class Builder {
       // 2. `name(args) => {code}`
       // 3. `(args) => {}`
       // 4. `args => {}`
+      // 5. `async name(args) => {code}`
+      // 6. `async (args) => {}`
+      // 7. `async args => {}`
       const isArrowWithoutParens = /^[a-zA-Z0-9_]+\s*=>/i.test(fnStr);
       const appendFunction =
-        !fnStr.startsWith('function') && !fnStr.startsWith('(') && !isArrowWithoutParens;
+        !fnStr.startsWith('function') &&
+        !fnStr.startsWith('async') &&
```

Comment:
great find! can we add an e2e test for this?

Reply:
might be hard to write an e2e test for this given that it's serializing functions sent to the editor, but at the very least we should write some unit tests for `serializeFn` with all the different inputs we expect:
- starts with function
- starts with async
- with parens
- without parens